src/main/java/com/exactpro/th2/conn/MicroserviceMain.java

```diff
@@ -215,30 +214,35 @@ private static Object getParamValue(ISettingsProxy settings, String parameterNam
         LOGGER.info("AvailableProcessors '{}'", Runtime.getRuntime().availableProcessors());
 
         return flowable
-                .doOnNext(message -> LOGGER.trace(
-                        "Message before observeOn with sequence {} and direction {}",
-                        message.getSequence(),
-                        message.getDirection()
-                ))
+                .doOnNext(message -> {
+                	LOGGER.trace(
+							"Message before observeOn with sequence {} and direction {}",
+							message.getSequence(),
+							message.getDirection()
+					);
+				})
                 .observeOn(PIPELINE_SCHEDULER)
                 .doOnNext(connectivityMessage -> LOGGER.debug("Start handling connectivity message {}", connectivityMessage))
-                .groupBy(ConnectivityMessage::getDirection)
+				.groupBy(ConnectivityMessage::getSessionAlias)
                 .map(group -> {
-                    @NonNull Direction direction = requireNonNull(group.getKey(), "Direction can't be null");
+                	AtomicReference<@NonNull Direction> direction = new AtomicReference<>();
```

Comment:
Let's get rid of this and just insert subscription for SECOND direction into the pipeline and will check the direction inside the subscribe method. If direction is not SECOND we can simply return from the method and wait for another message